### PR TITLE
consume child artifacts only

### PR DESCRIPTION
**Pull request by maggu2810**
Sorry, but my PR introduced a bug.
I will add further testing, but we / all need to be informed.

Sorry!

### CODE REVIEW

**Comment by maggu2810** [comment]
This PR is not ready to be merged!

File: de.dentrassi.pm.maven/src/de/dentrassi/pm/maven/internal/MavenRepositoryChannelAggregator.java [hunk 1]
```diff
@@ -253,12 +253,12 @@ private String makeUrl ( final String channelId )
         {
             poms.add ( art );
         }
-        else
```
Comment (maggu2810) [inline de.dentrassi.pm.maven/src/de/dentrassi/pm/maven/internal/MavenRepositoryChannelAggregator.java]: If we are using children only, the else is not needed anymore.

File: de.dentrassi.pm.maven/src/de/dentrassi/pm/maven/internal/MavenRepositoryChannelAggregator.java [hunk 2]
```diff
@@ -253,12 +253,12 @@ private String makeUrl ( final String channelId )
...
+        for ( final String childId : art.getChildIds () )
```
Comment (maggu2810) [inline de.dentrassi.pm.maven/src/de/dentrassi/pm/maven/internal/MavenRepositoryChannelAggregator.java]: Let's consume children only!

**Comment by ctron** [comment]
I am not sure I understand what goes wrong.

**Comment by ctron** [comment]
Ok .. I think I know what goes wrong ... 

`findChildPoms` should not return the artifact itself, even if it is a POM file, since this will be checked before in in `getInfos.

Is that correct?

**Comment by maggu2810** [comment]
findChildPoms should return only pom of childs (and sometimes themselves) but not all POMs.

The iteration over all map values without filter for child IDs was a mistake.

**Comment by ctron** [comment]
Ok, I see. But should the method really return itself? Isn't that covered by the previous check in getInfo?

**Comment by maggu2810** [comment]
You are right.

We could remote "add myself" in findChildPoms or we could remove multiple lines in getInfo.

At the moment getInfo uses the same logic for themselves as for all childs (in a loop).
As we support multiple POMs for an artifact now, we could cleanup getInfo if we define findChildPoms returns themselves too (perhaps using an boolean argument to control this).

**Comment by maggu2810** [comment]
Just commit the changes so you see what I mean.
Will have to test it tomorrow.

**Comment by ctron** [comment]
I am not sure if I understand what the boolean parameter is for.

The longer I look at it, the more I think it would be a good idea to do a different approach.

* Keep `getInfo` and add a new `findPomArtifacts` method which returns `Collection<ArtifactInformation>`
* In `findPomArtifacts` create a list of type `Collection<ArtifactInformation>`
 * Call a new method `fillFromArtifact`, which receives this list as input parameter and adds the artifact if it seems to be a pom file - actually the first check right now
 * Call a new method `fillFromChildren`, which again receives the same list as input parameter and adds all relevant children - actually the for look in `findChildPoms`
* Continue processing in `getInfo` as right now

This would allow use to easily customize the logic later on based on some channel configuration, maybe adding a parameter if child artifacts should be considered or not.

**Comment by maggu2810** [comment]
> I am not sure if I understand what the boolean parameter is for.

If the boolean parameter is
* true, findChildPoms act as fillFromArtifact + fillFromChildren
* false, findChildPoms act as fillFromChildren

I could split findChildPoms into findPomArtifacts, fillFromArtifact and fillFromChildren. Semms ok for me, too.

The coding part will be done tomorrow.

**Comment by ctron** [comment]
Ok, sounds great. I would like to get out 0.9.0 today or tomorrow if it is possible. Because I will have no time on Friday for it. Would this be possible?

**Comment by maggu2810** [comment]
I think so ;-)

**Comment by maggu2810** [comment]
Could you have a look at the code, whether this corresponds to your imagination.

**Comment by ctron** [comment]
Looks good to me. I wait what travis has to say about it and merge.

So I can try to make the 0.9.0 build today.

**Pull request merged**